index.d.ts

```diff
@@ -24,626 +24,6 @@ declare global {
 
   ///////////////////////////////////////////////////////////////////////////////
   // Widget
-
-  interface WidgetAPI {
-    register(component: FunctionalWidget<any>): void
-    h(...args: any[]): FigmaDeclarativeNode
-
-    // Hooks
-
-    /**
-     * @deprecated use useWidgetNodeId instead
-     */
-    useWidgetId(): string
-    useWidgetNodeId(): string
-    useSyncedState<T = any>(
-      name: string,
-      defaultValue: T | (() => T),
-    ): [T, (newValue: T | ((currValue: T) => T)) => void]
-    useSyncedMap<T = any>(name: string): SyncedMap<T>
-    usePropertyMenu(
-      items: WidgetPropertyMenuItem[],
-      onChange: (event: WidgetPropertyEvent) => void | Promise<void>,
-    ): void
-
-    useEffect(effect: () => (() => void) | void): void
-
-    useStickable(onStuckStatusChanged?: (e: WidgetStuckEvent) => void | Promise<void>): void
-    useStickableHost(
-      onAttachmentsChanged?: (e: WidgetAttachedStickablesChangedEvent) => void | Promise<void>,
-    ): void
-
-    waitForTask(promise: Promise<any>): void
-
-    // Components
-    AutoLayout: AutoLayout
-    Frame: Frame
-    Image: ImageComponent
-    Rectangle: Rectangle
-    Ellipse: Ellipse
-    Text: TextComponent
-    SVG: SVG
-    Input: InputComponent
-    Line: Line
-    Fragment: Fragment
-    Span: Span
-  }
-
-  interface WidgetStuckEvent {
-    // This is the id of the new node that your widget is stuck to
-    // or null if it is no longer stuck to anything
-    newHostId: string | null
-    // This is the id of the node that your widget was stuck to or null if it isn't stuck to anything
-    oldHostId: string | null
-  }
-
-  interface WidgetAttachedStickablesChangedEvent {
-    stuckNodeIds: string[]
-    unstuckNodeIds: string[]
-  }
-
-  type SyncedMap<T = any> = {
-    /** @deprecated use size instead */
-    readonly length: number
```

Comment:
lets keep this around too so we remain backwards compatible with the existing typings

if we remove this, code that uses these will start to fail typecheck when they update typing versions